changes for linux compatibility

rfhb opened the pull request:
Hi and thanks for the package! 
Attempting to install on Linux, the errorr is 

> In file included from xattrs-main.cpp:5:0:
> extattr.h: In function ‘bool removexattr(std::__cxx11::string, int)’:
> extattr.h:269:37: error: ‘listxattr’ was not declared in this scope
>  attr_names_t names = listxattr(path);
(multiply repeated)

For Linux, I added `#if defined()` sections complementary to those for macOS (where I could install without problems) and FreeBSD, and I also adapted function parameters. The package now compiles and loads under Linux - but works only partly under R: 

- working: has_xattrs(), list_xattrs(), get_attr() 
- no effect: set_xattr()
- error: rm_xattr() does not finish

I wondered if the PR helps to progress with the package; later I might further contribute depending on my C++.

rfhb committed:
changed for linux compatibility

hrbrmstr commented:
This is awesome! Can you amend the PR to add yourself as aut/ctb to the DESCRIPTION as well? #ty

rfhb committed:
keep tests that are working under linux, add reading attributes set by command tool

rfhb committed:
replacing included header files

rfhb committed:
accepted invitation to be listed

rfhb committed:
main linux adaptation: mangling of atrtibute names which under linux (only?) require a namespace as prefix

rfhb committed:
have has_xattrs() defined first avoids compilation errors

rfhb committed:
change R functions get_xattr() and set_xattr() to mangle attribute name for linux

rfhb committed:
change C functions get_xattr() and list_xattrs() into R functions to permit attribute name mangling for linux

rfhb committed:
autogenerated by Rcpp::compileAttributes()

rfhb committed:
added invisible return values for execution of functions set_xattr() and rm_xattr() to permit error checking

rfhb committed:
extended testing including attribute handling using OS tools

rfhb committed:
correct testing for macOS

rfhb committed:
quote strings for testing for macOS

Merge branch 'master' of https://github.com/rfhb/xattrs

# Conflicts:
#	tests/testthat/test-xattrs.R

rfhb commented:
Many thanks for the invite. 
For linux, mangling of attribute names seems required (e.g. user.attributename).
To demo the required changes, I have changed these functions to work with linux: 
- has_xattrs()
- list_xattrs()
- get_attr()
- set_attr()
- rm_attr()

If approach is ok, further work is to be done for most other functions. 
Along the way, corrected rm_attr() and extended testing. 
The package currently compiles and successfully tests under Linux and macOS.

hrbrmstr commented:
This is 👍🏼 on so many levels (I'm elated someone besides me pokes at xattrs!)

That approach is 100% OK.

_thank you!!_
